### PR TITLE
Move from travis to github actions

### PR DESCRIPTION
**Pull request by longwuyuan**
This is my first ever PR in life. I am just heavily inclined to contribute and this seemed like easy-peazy CI work. I will change/alter/learn/revert based on feedback. Sincere apologies if I cause any trouble here. Fingers crossed

 https://github.com/kubernetes-sigs/yaml/issues/48

### CODE REVIEW

**Comment by longwuyuan** [comment]
Reply after CLA

File: .travis.yml [hunk 1]
```diff
@@ -10,4 +10,4 @@ script:
   - GO111MODULE=on go test -v -race ./...
   - git diff --exit-code
 install:
-  - GO111MODULE=off go get golang.org/x/lint/golint
+  - GO111MODULE=off go get golang.org/x/lint/golin
```
Comment (palnabarun) [inline .travis.yml:13]: Is this change required?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: Go
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
```
Comment (palnabarun) [inline .github/workflows/go.yml:10]: nit: extra empty line

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  build:
+    strategy:
+      matrix: 
+        go-version: [1.12.x, 1.13.x, 1.14.x]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
```
Comment (palnabarun) [inline .github/workflows/go.yml:18]: nit: extra empty line

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    - name: Run golint
+      run: diff -u <(echo -n) <(golint $(go list -e ./...) | grep -v YAMLToJSON)
+    - name: Run go vet
+      run: GO111MODULE=on go vet .
+    - name: Run go test
+      run: GO111MODULE=on go test -v -race ./...
+    - name: Check diff
+      run: git diff --exit-code
+    - name: Install
+      run: GO111MODULE=off go get golang.org/x/lint/golint
```
Comment (palnabarun) [inline .github/workflows/go.yml:34]: This is a dependency of the whole process. Please move it to right after `Setup Go`.

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+    strategy:
+      matrix: 
+        go-version: [1.12.x, 1.13.x, 1.14.x]
```
Comment (palnabarun) [inline .github/workflows/go.yml:14]: Just curious, are we also adding Go 1.14.x to the build matrix?

Comment (palnabarun) [inline .github/workflows/go.yml:14]: The variable should be either be `go-versions` as used below in [L22](https://github.com/kubernetes-sigs/yaml/pull/49/files#diff-678682767f2477de3e3c546746f8568b9a1942b2c647d32331d7e774b8ff8d9fR22) or the variable name in that line needs to be changed.

```suggestion
        go-versions: [1.12.x, 1.13.x, 1.14.x]
```

**Review by palnabarun — CHANGES_REQUESTED** [review]
@longwuyuan -- Thank you for working on this and congratulations on your first PR! 🎉

File: .travis.yml (hunk 1 above)
Comment (longwuyuan) [reply .travis.yml:13]: Repo is ok .. my  workspace has the typo.. checking why

File: .github/workflows/go.yml (hunk 2 above)
Comment (longwuyuan) [reply .github/workflows/go.yml]: Had used the github gui to init so the emptylines came with the template I suspect. removed empty line

File: .github/workflows/go.yml (hunk 3 above)
Comment (longwuyuan) [reply .github/workflows/go.yml]: Had used the github gui to init so the emptylines came with the template I suspect. removed empty line

File: .github/workflows/go.yml (hunk 4 above)
Comment (longwuyuan) [reply .github/workflows/go.yml]: Saw the wierdness that linter installation came after linting...and now on second thought should have flagged it ... (first pr ever so did not want to question as this seemed like a working sequence. Now seems like travis has may have special treatment for the name "Install". Moved to beginning now

File: .github/workflows/go.yml (hunk 5 above)
Comment (longwuyuan) [reply .github/workflows/go.yml:12]: @dims showed klog repo which uses 1.14 and kinda seemed best practice to build for 1 up and one down with 13 being the middle as several folks may be around 14 and 15 or 15+ would be bleeding-edge. Should i remove it ? 1.14 was not in the original travis yaml

Comment (longwuyuan) [reply .github/workflows/go.yml:12]: i thought i used the plural in both places. Fixed now

**Comment by dims** [comment]
@longwuyuan also please squash the commits into a single commit, see how to here:
https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec

**Comment by longwuyuan** [comment]
> @longwuyuan -- Thank you for working on this and congratulations on your first PR!

:-) Really a big huge deal for me... looking to see how it goes and in any case ... my first PR .. and that too into k8s ...:-)

**Comment by longwuyuan** [comment]
squashing commits now

**Comment by longwuyuan** [comment]
@dims delayed as had to really learn (jitters/unaware over the -f for push).
Attempted it so look forward to feedback.
Absolutely delighted and grateful @palnabarun

File: yaml_test.go [hunk 6]
```diff
@@ -428,6 +428,7 @@ foo: baz
 }
 
 func TestJSONObjectToYAMLObject(t *testing.T) {
+	const bigUint64 = ((uint64(1) << 63) + 500) / 1000 * 1000
```
Comment (dims) [inline yaml_test.go:431]: you should not be touching yaml_test.go in your PR @longwuyuan please drop the commit that does this.

File: .travis.yml (hunk 1 above)
Comment (dims) [reply .travis.yml:13]: This is still broken should be `golint` not `golin`

**Comment by longwuyuan** [comment]
I will close this PR and submit a new one so as to not to complicate this, while I have the option. My rebasing knowledge is not up to par but good I now know that so won't repeat these mistakes. Thanks

**Comment by longwuyuan** [comment]
Although I have closed this PR, I am updating this for posterity sake.
Due to some git or linux-desktop related unknown funk, I repeatedly got a truncated file, on pulling from github and it resulted in yet another aspect of the broken PR. @dims made me aware of that with below comment in screenshot
image

But this  is closed anyways and I have a new PR submitted where I did not experience this truncated file problem. I will read up if any known precedents of truncated files etc.
